src/main/kotlin/biokotlin/util/MergeGVCFUtil.kt

```diff
@@ -0,0 +1,361 @@
+                            // creating the VariantContext
+                            variant.isRefBlock -> {
+                                variantsUsed.add(variant)
+                                val ploidy = variant.genotypeStrs(0).size
+                                Pair(variant.isPhased(0), MutableList(ploidy) { "REF" })
+                            }
+
+                            // Don't think this will be executed, as positions that have
+                            // indels will not be processed by this method
+                            else -> {
+                                Pair(false, listOf(".")) // No call
+                            }
+
+                        }
+
+                    } else { // Current variant for this sample doesn't represent the current position
+                        Pair(false, listOf(".")) // No call
+                    }
+                }
+            }
+        }
+
+    return createVariantContext(
+        VariantContextInfo(
+            currentPosition,
+            refAllele ?: error("Reference allele is null"),
+            samples,
+            altAlleles,
+            genotypes,
+            variantsUsed
+        )
+    )
+
+}
+
+private data class VariantContextInfo(
+    val position: Position,
+    val reference: String,
+    val samples: List<String>,
+    val altAlleles: Set<String>,
+    val genotypes: List<Pair<Boolean, List<String>>>, // Pair<phased, alleles>
+    val variantsUsed: List<SimpleVariant>
+)
+
+/**
+ * Creates a VariantContext for the current position.
+ */
+private fun createVariantContext(info: VariantContextInfo): VariantContext {
+
+    val refAllele = alleleRef(info.reference)
+
+    val alleleMap = mutableMapOf<String, Allele>()
+    alleleMap[info.reference] = refAllele
+    info.altAlleles.forEach { alleleMap[it] = alleleAlt(it) }
+
+    val genotypes = info.genotypes.mapIndexed { index, (phased, alleles) ->
+
+        val alleleObjs = alleles
+            .map { allele ->
+                when (allele) {
+                    "." -> Allele.NO_CALL
+
+                    "REF" -> refAllele
+
+                    else -> alleleMap[allele] ?: throw IllegalArgumentException("Allele not found: $allele")
+                }
+            }
+
+        GenotypeBuilder(info.samples[index], alleleObjs)
+            .phased(phased)
+            .make()
+
+    }
+
+    return VariantContextBuilder()
+        .source(".")
+        .alleles(alleleMap.values)
+        .chr(info.position.contig)
+        .start(info.position.position.toLong())
+        .stop(info.position.position.toLong())
+        .genotypes(genotypes)
+        .make()
+
+}
+
+var timeMinPosition = 0L
+var timeNextLowestPosition = 0L
+var timeLowestLookAheadStart = 0L
+var timeResult = 0L
+var timeAdvanceVariant = 0L
+private fun nextPosition(gvcfReaders: Array<VCFReader>, currentPosition: Position? = null): Position? {
```

Comment:
Add a comment here explaining what this is doing so it is more understandable.  Also consider removing timing code as it might be complicating things.

Reply:
That method was replaced by nextPositions() when I refactored for performance. But I did add comments for nextPositions(). And the timing code was removed when I refactored